lisa/base_tools/apt_add_repository.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import cast
+
+from lisa.executable import Tool
+
+
+class AptAddRepository(Tool):
+    @property
+    def command(self) -> str:
+        return "apt-add-repository"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        from lisa.operating_system import Ubuntu
```

Comment:
put it on the top

Reply:
In base_tools, it cannot reference os. This line should be moved to above, and in type checking condition. From this place, use a line like below to workaround it.

```python
ubuntu: Ubuntu = self.node.os # type: ignore
```

Reply:
Thank you for the pointer, I was not sure how to do it because I found multiple examples throughout the repo.